File: js/euler26.js
function euler26(){
    var res = 0, maxLen = 0;
    for (let n = 2; n < 1000; n++){
        var len = getCycle(n);
        if (len > maxLen){
            maxLen = len;
            res = n;
        }
    }
    return res;
}

function getCycle(n){
    var rem = 1, count = 0;
    var seen = new Map();
    while (rem != 0){
        rem = (rem * 10) % n;
        if (seen.has(rem)) return (count - seen.get(rem));
        seen.set(rem, count++);
    }
    return 0;
}

console.log(euler26());

File: go/euler26.go
package main
import "fmt"

func getCycleLength(n int) int {
    seen := make(map[int]int)
    rem := 1
    curr := 0
    for rem != 0 {
        rem = (rem * 10) % n
        _, found := seen[rem]
        if found {
            return curr - seen[rem]
        }
        seen[rem] = curr
        curr++
    }
    return 0
}

func main() {
    res, maxCycle := 0, 0
    for d := 2; d < 1000; d++ {
        cycle := getCycleLength(d)
        if cycle > maxCycle {
            res = d
            maxCycle = cycle
        }
    }
  fmt.Println(res)
}
